systems-state-manager/nisystemsstate.yml

```diff
@@ -0,0 +1,999 @@
+    type: string
+    enum:
+      - ARM
+      - X64
+      - X86
+    example: ARM
+
+  StateDistribution:
+    description: Supported distribution by a state.
+    title: State Distribution    
+    type: string
+    enum:
+      - NI_LINUXRT
+      - NI_LINUXRT_NXG
+      - WINDOWS
+    example: NI_LINUXRT
+
+  StateMetadata:
+    description: State metadata information.
+    title: State Description
+    type: object
+    properties:
+      id:
+        type: string
+        description: The ID of the state.
+        example: 5bc4463af0f06f4585003ae6
+      name:
+        type: string
+        description: The name of the state.
+        example: crio9064State
+      fileName:
+        type: string
+        description: The file name of the state.
+        example: crio9064State
+      description:
+        type: string
+        description: The description of the state.
+        example: crio9064 Setup State
+      distribution:
+        $ref: '#/definitions/StateDistribution'
+      architecture:
+        $ref: '#/definitions/StateArchitecture'
+      createdTimestamp:
+        description: ISO-8601 formatted timestamp specifying the state creation date.
+        type: string
+        format: iso-date-time
+        example: '2018-05-07T18:58:05.219692Z'
+      lastUpdatedTimestamp:
+        description: ISO-8601 formatted timestamp specifying the last date that the state has been updated.
+        type: string
+        format: iso-date-time
+        example: '2018-05-07T18:58:05.219692Z'
+    required:
+      - id
+      - name
+      - fileName
+      - distribution
+      - architecture
+      - createdTimestamp
+      - lastUpdatedTimestamp
+
+  StateDescriptionListResponse:
+    description: The response containing a list of state metadata.
+    title: State Description List Response
+    type: object
+    properties:
+      totalCount:
+        type: integer
+        description: Total count of states.
+      states:
+        type: array
+        items:
+          $ref: '#/definitions/StateMetadata'
+    example:
+      totalCount: 2
+      states:
+        - id: '5bc5e9092a4fa4c71cfa7197'
+          name: vim
+          description: Having Vim installed on the windows system.
+          distribution: WINDOWS
+          architecture: x64
+          createdTimestamp: '2018-05-07T18:58:05.219692Z'
+          lastUpdatedTimestamp: '2018-05-07T18:58:05.219692Z'
+        - id: '5bcde471fc13379595ce9b13'
+          name: emacs
+          description: Having Emacs installed on the windows system.
+          distribution: WINDOWS
+          architecture: x86
+          createdTimestamp: '2018-05-07T18:58:05.219692Z'
+          lastUpdatedTimestamp: '2018-05-07T18:58:05.219692Z'
+    required:
+      - totalCount
+      - states
+
+  StateRequest:
+    description: System state object.
+    title: State Request
+    type: object
+    properties:
+      name:
+        type: string
+        description: Name of the State
```

Comment:
More of a question about consistency. I see that these descriptions don't have periods and that "state" is capitalized. However, the properties descriptions following this request's either have periods or don't capitalize "state". Should we add periods here and lowercase "state" here and in the PatchStateRequest's properties' descriptions?